sql/core/src/main/scala/org/apache/spark/sql/execution/WindowExec.scala

```diff
@@ -625,10 +643,12 @@ private[execution] final class OffsetWindowFunctionFrame(
     if (inputIndex >= 0 && inputIndex < input.size) {
```

Comment:
This is a more general comment, which does not necessarily apply to this line. Since we are breaking the code up into to separate code paths (with row/without row), we might as well get rid of the joined row and the logic needed to set this up (like: `Seq.fill(ordinal)(NoOp)`)